kytos/core/tcp_server.py

```diff
@@ -44,113 +40,108 @@ def __init__(self, server_address, RequestHandlerClass,
         """
         super().__init__(server_address, RequestHandlerClass,
                          bind_and_activate=False)
-        # Registering the register_event 'endpoint' on the server to be
-        #   accessed on the RequestHandler
         self.controller = controller
-        # self.controller_put_raw_event = controller_put_raw_event
 
     def serve_forever(self, poll_interval=0.5):
-        """Handle requests until an explicit shutdown() request."""
+        """Handle requests until an explicit shutdown() is called."""
+        SOCKET_BIND_WAIT_TIME = 0.5
+        SOCKET_BIND_N_TRIES = 2
+        # Tries to bind the socket n times, it may still be
+        # in WAIT state from previous connection.
+        for i in range(SOCKET_BIND_N_TRIES):
```

Comment:
We ( @erickvermot , @abaruchi , @renanrodrigo and I) discussed this and the possibility of errors due to reusing the connection, considering the OpenFlow protocol (or eventually any other SDN protocol) is negligible. Even if it occurs, the consequences are minimal and manageable, in a way that the costs of waiting the `TIME_WAIT` to expire seems to be much higher than just reusing the socket and eventually dropping some lost packages or restarting the connection to a switch.

So, let's stick with the option 3 and reuse the socket without waiting the `TIME_WAIT`.